### DISCUSSION TITLE
Match imports with a Lookup text instead of ID

### DISCUSSION THREAD

**Discussion by imranakram** [rappen/Xrm.Shuffle#6] [Q&A]
A basic question about Shuffle data import. 

How does the mapping of lookup work when using a lookup in a matching criterion? If we want to match on the lookup field's display name

For example this import block:-

```
<Import CreateWithId="true" >
     <Match PreRetrieveAll="true" >
         <Attribute Name="cint_union_id" />
          <Attribute Name="contactid" />
          <Attribute Name="cint_union_section_id" />
    </Match>
</Import>
```

The lookup cint_union_section_id is trying to match on the guid it seems.
![image](https://github.com/rappen/Xrm.Shuffle/assets/13014005/99d065fb-7edd-4326-b268-eb14471bb59b)

**Comment by rappen** [comment]
First - it matches with the related's Id.

Second - if you export the related records, and import them (without even importing them!) you get the mapping between Ids in the source environment and the target environment.

Example of an extremely simple Shuffle:
```
<ShuffleDefinition>
  <Blocks>
    <DataBlock Name="Account" Entity="account" Type="Entity" >
      <Export>
        <Attributes>
          <Attribute Name="name" />
          <Attribute Name="accountnumber" />
        </Attributes>
      </Export>
      <Import Save="Never" >
        <Match>
          <Attribute Name="accountnumber" />
        </Match>
      </Import>
    </DataBlock>
    <DataBlock Name="Contact" Entity="contact" Type="Entity" >
      <Export>
        <Attributes>
          <Attribute Name="parentcustomerid" />
          <Attribute Name="firstname" />
          <Attribute Name="lastname" />
        </Attributes>
      </Export>
      <Import>
        <Match>
          <Attribute Name="parentcustomerid" />
          <Attribute Name="firstname" />
        </Match>
      </Import>
    </DataBlock>
  </Blocks>
</ShuffleDefinition>
```
Connecting contact to the correct account, without importing the accounts.

Makes sense?
